aioftp/client.py

```diff
@@ -547,13 +697,38 @@ def list(self, path="", *, recursive=False):
 
             >>> stats = await client.list()
         """
-        class AsyncClientLister(AsyncListerMixin):
+        class AsyncListerClient(AsyncListerMixin):
 
             async def _new_stream(cls, local_path):
 
                 cls.path = local_path
-                command = str.strip("MLSD " + str(cls.path))
-                return await self.get_stream(command, "1xx")
+                str_path = " " + str(cls.path)
+                if self.list_compatibility_level > 0:
+
+                    cls._anext = cls._anext_MSLD_LIST
+
+                    if self.list_compatibility_level == 2:
+
+                        cls.process_line = self.parse_mlsx_line
+
+                        try:
+                            command = str.strip("MLSD" + str_path)
+                            return await self.get_stream(command, "1xx")
+
+                        except errors.StatusCodeError as e:
+
+                            if e.received_codes[-1] == "500":
+
+                                self.list_compatibility_level = 1
+
+                    if self.list_compatibility_level == 1:
+
+                        cls.process_line = self.parse_list_line
+                        command = str.strip("LIST" + str_path)
+                        return await self.get_stream(command, "1xx")
+                else:
+                    # TODO: implement NLST/CWD fallback
+                    pass
```

Comment:
* If the command exists, but we failed on parsing, then we should raise exception, no fallback.
* network traffic is about performance, and this one is pretty cheap. So, no caching right now.